velox/dwio/common/Options.h

```diff
@@ -605,10 +606,13 @@ struct WriterOptions {
   std::optional<bool> orcWriterIntegerDictionaryEncodingEnabled{std::nullopt};
   std::optional<bool> orcWriterStringDictionaryEncodingEnabled{std::nullopt};
   std::map<std::string, std::string> serdeParameters;
-  std::optional<uint8_t> parquetWriteTimestampUnit;
   std::optional<uint8_t> zlibCompressionLevel;
   std::optional<uint8_t> zstdCompressionLevel;
 
+  // Writer implementations should provide this function to specify how to
+  // process writer-specific input session and connector configs.
+  virtual void processUserConfigs(const Config*) = 0;
+
   virtual ~WriterOptions() = default;
```

Comment:
I would prefer composition over inheritance for such POD object.  That way `processUserConfigs` can be hidden in the format specific code and not exposed.

Reply:
Discussed offline that this object will no longer be POD so we just keep the inheritance here.